docs/src/model/md-generate.tsx

```diff
@@ -0,0 +1,115 @@
+import path from "path";
+import glob from "glob";
+import fs from "fs";
+
+export const DOCS_PATH = path.join(process.cwd(), "src/content/");
+export const newLineDividerRegEx = /\r\n|\n/;
+
+export const withIndexURL = (pathname: string): string[] => {
+  const docPath = pathname.split("/");
+  if (docPath[docPath.length - 1] === "index") {
+    return [...docPath.slice(0, docPath.length - 1)];
+  }
+  return docPath;
+};
+
+export const withIndexFile = (docPath: string[], isIndex: boolean): string => {
+  const mdFilePath = path.join(
+    DOCS_PATH,
+    `${docPath.join("/")}${isIndex ? "/index" : ""}.md`
+  );
+  return mdFilePath;
+};
+
+export const withCodeElementWrapper = (content: string) =>
+  `
+   \`\`\` 
+    ${content}   
+    \`\`\`
+  `;
+
+export const getEntriesInfo = (
+  line: string
+): {
+  pathname: string;
+  rowsNumbers: [number, number] | null;
+} => {
+  const rowsNumbers: [number, number] | null = line.includes("{")
+    ? (line
+        .substring(line.indexOf("{"))
+        .replace(/[{}]/g, "")
+        .split("-")
+        .map((lineNumberString) => Number(lineNumberString)) as [
+        number,
+        number
+      ])
+    : null;
+
+  const pathname = (
+    rowsNumbers ? line.substring(0, line.indexOf("{")) : line
+  ).replace("<<< @/", "");
+
+  return {
+    pathname,
+    rowsNumbers,
+  };
+};
+
+export const getContentFromRange = (
+  content: string,
+  rowsNumbers: [number, number]
+) => {
+  const [startLineNumber, endLineNumber] = rowsNumbers;
+
+  return content
+    .split(newLineDividerRegEx)
+    .slice(startLineNumber, endLineNumber)
+    .join("\n");
+};
+
+export const withInsertedCodeFromLinks = (content: string) => {
+  return content
+    .split(newLineDividerRegEx)
+    .map((line: string) => {
+      if (!line.startsWith("<<<")) return line;
```

Comment:
This is looks good for me now. Let's add two small editions:
1 handle cases with wrong links. e.g. what if file doesn't exist
2 can we refactor a bit to have only single line `return withCodeElementWrapper(contentFromRange);` but select contentFromRange based on parameters?

Reply:
done